src/Provider/Google.php

```diff
@@ -117,6 +123,16 @@ protected function checkResponse(ResponseInterface $response, $data)
 
     protected function createResourceOwner(array $response, AccessToken $token)
     {
-        return new GoogleUser($response);
+        $user = new GoogleUser($response);
+        // Validate hosted domain incase the user edited the initial authorization code grant request
+        if ($this->hostedDomain === '*') {
+            if (empty($user->getHostedDomain())) {
+                throw new HostedDomainException($this->hostedDomain, $user->getHostedDomain());
+            }
```

Comment:
This implies a significant change to current functionality, requiring all users to set `hostedDomain`. I think this check should be skipped unless `hostedDomain` option has been set, at least until the next major version.

Reply:
I think I'm missing your point - this only gets checked if `hostedDomain` is set to '*' or is not empty, so it doesn't require everyone to configure a `hostedDomain`.

It does affect everyone with `hostedDomain` configured though I expect those people don't realize how easily that can be bypassed and already assume the library validates it.